ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/LogSegment.java

```diff
@@ -452,9 +453,17 @@ public String toString() {
         "log-" + startIndex + "_" + endIndex;
   }
 
-  @Override
-  @SuppressFBWarnings("EQ_COMPARETO_USE_OBJECT_EQUALS")
-  public int compareTo(Long l) {
+  /** Comparator to find <code>index</code> in list of <code>LogSegment</code>s. */
+  static final Comparator<Object> SEGMENT_TO_INDEX_COMPARATOR = (o1, o2) -> {
```

Comment:
The `Comparator<Object>` is a great idea!  I was thinking implementing binary search on our own.